http-server-netty/src/test/groovy/io/micronaut/http/server/netty/ports/ControllerPortSpec.groovy

```diff
@@ -0,0 +1,57 @@
+package io.micronaut.http.server.netty.ports
+
+import io.micronaut.context.ApplicationContext
+import io.micronaut.core.io.socket.SocketUtils
+import io.micronaut.http.HttpStatus
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.client.RxHttpClient
+import io.micronaut.http.client.exceptions.HttpClientResponseException
+import io.micronaut.runtime.server.EmbeddedServer
+import spock.lang.Retry
+import spock.lang.Specification
+
+class ControllerPortSpec extends Specification {
+
+
+    @Retry // try because a port binding issue could occur on CI
+    void "test custom controller port"() {
+
+        given:
+        def customPort = SocketUtils.findAvailableTcpPort()
+        EmbeddedServer embeddedServer = ApplicationContext.run(
+                EmbeddedServer,
+                ['my.controller.port': customPort]
+        )
+        def client = embeddedServer.applicationContext.createBean(RxHttpClient, new URL("http://localhost:$customPort"))
+        def client2 = embeddedServer.applicationContext.createBean(RxHttpClient, embeddedServer.getURL())
+
+        when:
+        def response = client.toBlocking().retrieve("/custom-port1")
+
+        then:
+        response == 'ok'
+
+        when:
+        client2.toBlocking().retrieve("/custom-port1")
+
+        then:
+        def e = thrown(HttpClientResponseException)
+        e.response.status() == HttpStatus.NOT_FOUND
+
+
+        cleanup:
+        embeddedServer.close()
+        client.close()
+        client2.close()
+    }
+
+    @Controller(value = "/custom-port1", port='${my.controller.port}')
```

Comment:
Shouldn't there be a test for two controllers with two different ports that requests to one doesn't get to a route the other one has? And what happens when a port isn't specified on a particular Controller? Does it default to port 80 or a random port?

Reply:
the test is already doing that be ensuring a 404 is returned for same route for the default port.

When a port is not specified it behaves the same as it does now